pages/bookshelf.tsx

```diff
@@ -0,0 +1,65 @@
+import { NextPage } from "next"
+import "react-toastify/dist/ReactToastify.css"
+import Navigation from "../components/bookshelf/Navigation"
+import { Container } from "../components/styled/bookshelf"
+import { useState, useContext, useEffect, useMemo } from "react"
+import Discover from "../components/bookshelf/discover/Discover"
+import LogoutBox from "../components/bookshelf/LogoutBox"
+import { UserContext } from "../context/UserContext"
+import { useRouter } from "next/router"
+import { getBooks } from "../api/books"
+import { BooksContext } from "../context/BooksContext"
+import { ActivePageEnum } from "../api/enums"
+
+const Home: NextPage = () => {
+  const [active, setActive] = useState<ActivePageEnum | string>(
+    ActivePageEnum.to_read
+  )
+  const router = useRouter()
+
+  useEffect(() => {
+    const existingUser = JSON.parse(localStorage.getItem("user") || "null")
+    existingUser ? setUser(existingUser) : router.push("/")
```

Comment:
I suggest creating a new component. Maybe `AuthorizedLayout`. Then this Layout will be the one that checks if the user is present or not. 

If not present, redirect to the login page.

Reply:
then your bookshelf page would look like this:

```
return (
<AuthorizedLayout>
  <BookshelfComponent />
</AuthorizedLayout>
)
```

Reply:
Hello! is this the correct way to implement your suggestion here? 
![image](https://user-images.githubusercontent.com/91857581/188637253-25138bd7-d26b-4633-9175-530141c5069b.png)
![image](https://user-images.githubusercontent.com/91857581/188637272-777b6e86-cf3e-4238-8035-23643aae8d93.png)

Still using a check on localStorage but I will try to implement the /whoami endpoint soon

Reply:
Yeah that looks fine. Have a look at this too - https://nextjs.org/docs/basic-features/layouts and see if you can make use of NextJS layout instead of a custom one :)